test/asyncio_tests/test_examples.py

```diff
@@ -1302,3 +1303,105 @@ def strptime(s):
         # Start Versioned API Example 8
         # 8
         # End Versioned API Example 8
+
+    @env.require_version_min(5, 0)
+    @asyncio_test
+    async def test_snapshot_query(self):
+        client = self.cx
+        if not env.is_replica_set and not env.is_mongos:
+            self.skipTest("Must be a sharded or replicaset")
+
+        self.addCleanup(client.drop_database, "pets")
+        db = client.pets
+        await db.drop_collection("cats")
+        await db.drop_collection("dogs")
+        await db.cats.insert_one(
+            {"name": "Whiskers", "color": "white", "age": 10, "adoptable": True}
+        )
+        await db.dogs.insert_one(
+            {"name": "Pebbles", "color": "Brown", "age": 10, "adoptable": True}
+        )
+        await wait_until(lambda: self.check_for_snapshot(db.cats), "success")
+        await wait_until(lambda: self.check_for_snapshot(db.dogs), "success")
+
+        # Start Snapshot Query Example 1
+
+        db = client.pets
+        async with await client.start_session(snapshot=True) as s:
+            adoptablePetsCount = 0
+            cursor = db.cats.aggregate(
+                [{"$match": {"adoptable": True}}, {"$count": "adoptableCatsCount"}], session=s
+            )
+            async for doc in cursor:
+                adoptablePetsCount = doc["adoptableCatsCount"]
+                break
+
+            cursor = db.dogs.aggregate(
+                [{"$match": {"adoptable": True}}, {"$count": "adoptableDogsCount"}], session=s
+            )
+            async for doc in cursor:
+                adoptablePetsCount += doc["adoptableDogsCount"]
+                break
```

Comment:
I think it would be more idiomatic to use `to_list` for these:
```
            docs = await db.cats.aggregate(
                [{"$match": {"adoptable": True}}, {"$count": "adoptableCatsCount"}], session=s
            ).to_list(None)
            adoptablePetsCount += docs[0]["adoptableCatsCount"]
            ...
```

Reply:
Done